app/src/main/kotlin/trees/dataBases/BST/flatFilesDB.kt

```diff
@@ -0,0 +1,50 @@
+package trees.dataBases.BST
+
+import org.apache.commons.csv.CSVFormat
+import org.apache.commons.csv.CSVParser
+import org.apache.commons.csv.CSVPrinter
+import trees.BSTree
+import trees.nodes.BSNode
+import java.io.File
+
+const val fileName = "./app/src/main/kotlin/trees/dataBases/BST/flatFile"
+val curFile = File(fileName)
+
+fun removeFile(file: File = curFile) {
+    file.delete()
+    file.createNewFile()
+}
+
+fun writeAllNodesToFile(node: BSNode<Int, String>?, tree: BSTree<Int, String>, file: File = File(fileName)) {
+    if (!file.exists()) {
+        file.createNewFile()
+    }
+    val stack = mutableListOf(node?.key)
+    file.bufferedWriter().use {
+        val csvPrinter = CSVPrinter(it, CSVFormat.DEFAULT)
+        while (stack.isNotEmpty()) {
+            val current = stack.removeLast()?.let {current -> tree.find(current) }
+            if (current?.left != null)
+                current.left?.key.apply(stack::add)
+            if (current?.right != null)
+                current.right?.key.apply(stack::add)
+            csvPrinter.printRecord("${current?.key}","${current?.value}")
```

Comment:
1. По заданию вам же вроде бы нужно было и координаты тоже сохранять.
2. Используйте автоформатирование IntelliJ IDEA (Ctrl + Alt + L) -- у вас с этим проблемы не только здесь:
```suggestion
            csvPrinter.printRecord("${current?.key}", "${current?.value}")
```